### ISSUE TITLE
Choice-frontier adapter 3-seed replication + pre-registered separation from the exact-eps-0.75 rung (GPU)

### ISSUE THREAD

**Issue by Sino-Huang** [Sino-Huang/multimodality_on_planning#138]
# Runbook: 3-seed replication of the #136 adapter + pre-registered separation from the exact-eps-0.75 rung (GPU)

> The author is away and will not answer. Never ask questions or wait for replies. Use the **blocked exit** (rule 8) only for failures this ticket does not cover.

## 0. Why (context only)

Review feedback on #136 (POSITIVE, D = +0.328 [+0.155, +0.508]):
- it rests on **one training seed**;
- the adapter's M1 0.349 sits **at** the exact-eps-0.75 rung (0.347), and only point estimates were compared.

To reach a stronger rating, the reviewer wants 3 seeds and a learned result that is separated from the eps-0.75 rung by an interval. This ticket trains seeds 29 and 71 with the **frozen #136 corpus and recipe, unchanged**, then evaluates all 3 seeds on the frozen #135 panel. #139 then re-evaluates all 3 seeds on fresh held-out panels.

## 1. Rules

1. `source ~/cd_vlaplan &&` before Python. Repo root `/data/scratch/projects/punim0478/sukaih/multimodality_on_planning`. Always pass `gh ... --repo Sino-Huang/multimodality_on_planning`.
2. Never read `data/bfws_phase_v1/fresh-test-manifest.jsonl` or `data/curriculum_pddl/*/dev/`.
3. Never modify frozen files:
   - `scripts/run_choice_frontier*.py` that already exist, and `scripts/analyze_choice_frontier_v2.py` / `_v3.py`
   - anything under `outputs/choice-frontier/{v1,o4,v2,v3}/`
   - `configs/experiments/choice-frontier{,-v2,-v3}/`
   - Copy instead.
4. **The corpus and recipe are frozen.** Read `outputs/choice-frontier/v3/preparation/` and `configs/experiments/choice-frontier-v3/{protocol,membership}.json` read-only. Assert that the #136 `membership_sha256` (`f5cc9b2a…c077`) matches. Only the training seed changes.
5. **Shared program ledger.** This ticket creates `docs/experiments/choice-frontier/schedule-v4.json` and `outputs/choice-frontier/v4/budget.json`: cap **24 GPU-h**, allocation `choice_frontier` 24, MASTER_PORT pool **18826-18829**, gpu_cutoff and handoff dates far enough ahead for admission. #139 uses the **same** ledger and schedule, so the scheduler never double-books a GPU. All GPU work goes through `scripts/run_expanded_study.py launch`. Use a distinct MASTER_PORT per concurrent job and at most one job per GPU.
6. Commit and push the protocol (step 2) **before any training launch**.
7. Skip formatters and project-wide test suites. Run only the named pytest and ruff commands.
8. **Blocked exit:**
   - commit and push what is done;
   - write `docs/experiments/choice-frontier/issue-138-closeout.md` with a `## Blocked` section;
   - comment `## #138 blocked …` on the issue and leave it OPEN.

## 2. Step 1 — protocol (commit before any GPU launch)

Write `docs/experiments/choice-frontier/issue-138-protocol.md` and `configs/experiments/choice-frontier-v4/seeds-protocol.json` with exactly these decisions:

- **Training:** seeds **29** and **71** × {best_first_add_greedy, best_first_add_w3} = 4 cells. Corpus, augmentation, sample order and recipe are byte-identical to #136 (only the seed changes), as are the audit-train checks. Seed 17 adapters are reused read-only from `outputs/choice-frontier/v3/training/<algorithm>/seed-17/final/`.
  - The sample order must be identical to #136 (`Random(order:17)`), so the seed changes **only** the LoRA init and dropout. If the #136 code ties sample order to the training seed, keep the #136 behaviour exactly and record which it is.
- **Evaluation:** learned_adapter for the new cells on the frozen #135 panel (`configs/experiments/choice-frontier-v2/membership.json`, membership_sha256 `b2d909ca…11cd07`). Use greedy decoding, inference seed 17, cap 2 × R_t (the #135 exact expansions), the same smoke gate as #136 and replay of every episode. Controls, the exact-eps ladder and pretrained_base are reused from #135, and the seed-17 learned episodes from #136.
- **Primary endpoint (the #136 rule, now with 3 seeds):**
  - D3 = mean over seeds {17, 29, 71} of M1(learned, s) − M1(random_valid).
  - CI: task-cluster bootstrap, `random.Random(133)`, 10,000 draws; resample tasks and average the seeds inside each draw.
  - Verdict (unchanged margins δ = 0.05, equivalence ±0.05):
    - POSITIVE: lo > 0, D3 ≥ 0.05 and **every** seed's point estimate > 0.
    - EQUIVALENT: −0.05 < lo and hi < 0.05.
    - NEGATIVE: hi < 0.
    - INCONCLUSIVE: otherwise.
- **Pre-registered separation test (new, co-primary):**
  - S = M1(learned, 3-seed mean) − M1(exact-eps-0.75), where eps-0.75 is itself averaged over its 5 #135 seeds.
  - Use a paired task-cluster bootstrap with the same settings.
  - Verdict: `SEPARATED_ABOVE` if lo > 0; `SEPARATED_BELOW` if hi < 0; otherwise `NOT_SEPARATED`.
  - Report per-seed S points as well. Also report the same test against exact-eps-0.50 (descriptive only).
- **Seed variance (descriptive):** per-seed M1 with CI, per-seed teacher agreement − chance with CI, per-seed last-label rate, and the between-seed SD of M1.
- **Budget:** estimate from #136 measured hours (train ≈ 1.89 GPU-h per cell, evaluation ≈ 0.5 GPU-h per cell) and write it in the protocol.

Commit and push. Post the hash on this issue.

## 3. Step 2 — code, schedule, jobs

- Copy `scripts/run_choice_frontier_v3.py` → `scripts/run_choice_frontier_v4_seeds.py`:
  - output root `outputs/choice-frontier/v4/seeds/`;
  - ledger and schedule paths point to v4;
  - the corpus and store are read from v3 read-only;
  - training and evaluation stages accept seeds 29/71;
  - fix the #136 completion-hook `EXPANDED_ATTEMPT_DIR` issue the same way #136's closeout describes.
- Create job files `configs/experiments/choice-frontier-v4/cfv4-train-<alg>-s<seed>-job.json` (max_seconds 18000) and `cfv4-evaluate-<alg>-s<seed>-job.json`.
- Commit and push before the first launch.

## 4. Step 3 — train, smoke, evaluate

- `init` the v4 ledger.
- Launch the 4 training cells two at a time (one per GPU).
- Relaunch infrastructure failures at most 2 times; never change the recipe.
- Run the smoke gate on the seed-29 adapters (the #132 definition). If it fails, verdict `SMOKE_FAIL`: close out and close the issue.
- Evaluate the 4 new cells on the #135 panel, then `finalize` (0 missing, 0 mismatch).

## 5. Step 4 — analysis

Write `scripts/analyze_choice_frontier_v4_seeds.py`. Import the M1–M4 functions from `scripts/analyze_choice_frontier_v2.py`; do not copy the formulas. Output `outputs/choice-frontier/v4/seeds/metrics/analysis.json` with these blocks:
- `primary` {D3, ci95, per_seed, verdict}
- `separation` {vs_eps_0.75: {S, ci95, per_seed, verdict}, vs_eps_0.50: {…}}
- `per_seed`
- `seed_variance`
- `ladder_position`

## 6. Step 5 — tests, closeout

- Add `tests/planning_benchmark/test_choice_frontier_v4_seeds.py`. It must test that the seed-averaged-inside-draw bootstrap equals a hand-computed small case, and that the separation verdict rule is applied at its boundaries.
- Run pytest on that file and ruff on the two new scripts.
- Write `docs/experiments/choice-frontier/issue-138-closeout.md`, with the same sections as `issue-136-closeout.md`: verdicts first, evidence index with the first 16 hex of each sha256, and limitations.
- Post the closing comment and close the issue:

```
## #138 closeout
Primary (3 seeds): <POSITIVE|EQUIVALENT|NEGATIVE|INCONCLUSIVE|SMOKE_FAIL>, D3 = <x> [<lo>, <hi>]; per-seed s17 <a>, s29 <b>, s71 <c>
Separation vs exact-eps-0.75: <SEPARATED_ABOVE|NOT_SEPARATED|SEPARATED_BELOW>, S = <x> [<lo>, <hi>]
Keys for #131 (outputs/choice-frontier/v4/seeds/metrics/analysis.json): primary.*, separation.*, per_seed.*, seed_variance
GPU: <x>/24 GPU-h shared v4 ledger (this ticket's jobs: <y>)
Commits: protocol <sha>, code <sha>, evidence <sha>
```

**Comment by Sino-Huang** [comment]
Protocol frozen and pushed before any GPU launch: f06c7ac (docs/experiments/choice-frontier/issue-138-protocol.md + configs/experiments/choice-frontier-v4/seeds-protocol.json). Recorded decision: the #136 code ties sample order to the training seed (order:{training_seed}); v4 pins order:17 so samples/order are byte-identical to #136 and the seed changes only LoRA init + dropout. Budget estimate 9.66 GPU-h expected (12.08 worst case) on the shared 24 GPU-h v4 ledger.

**Comment by Sino-Huang** [comment]
## #138 closeout
Primary (3 seeds): POSITIVE, D3 = 0.285 [0.140, 0.433]; per-seed s17 0.328, s29 0.229, s71 0.297
Separation vs exact-eps-0.75: NOT_SEPARATED, S = -0.041 [-0.187, 0.102]
Keys for #131 (outputs/choice-frontier/v4/seeds/metrics/analysis.json): primary.*, separation.*, per_seed.*, seed_variance
GPU: 10.11/24 GPU-h shared v4 ledger (this ticket's jobs: 10.11)
Commits: protocol f06c7ac, code 00ac5bf (analyzer+test dd6e458), evidence b47cbaf

Notes: between-seed SD of M1 0.051 (0.349 / 0.250 / 0.318); vs exact-eps-0.50 (descriptive) S = -0.298 [-0.468, -0.129] SEPARATED_BELOW. Smoke (seed-29) PASS 31/31; finalize 48/48 replayed, 0 missing, 0 mismatch. One infrastructure relaunch (cfv4-evaluate-best_first_add_w3-s29: attempt 1 launched without ~/cd_vlaplan env, failed at tokenizer load before any episode, 0.006 GPU-h). Sample order pinned to #136 order:17 for all seeds (recorded in protocol). Adapters for #139: outputs/choice-frontier/v4/seeds/training/<algorithm>/seed-<29|71>/final/ (seed 17: outputs/choice-frontier/v3/training/<algorithm>/seed-17/final/). Closeout: docs/experiments/choice-frontier/issue-138-closeout.md
